tools/seed_table.cpp

```diff
@@ -168,53 +168,63 @@ struct GameState {
 	uint32_t startingSeed;
 
 	std::array<int32_t, 17> seedTable {};
+	std::array<int32_t, 17> stateTable {};
 
 	GameState() = delete;
 
 	GameState(uint32_t state)
 	    : startingSeed(state)
 	{
-		for (auto &dungeonSeed : seedTable) {
-			state       = advanceRng(state);
-			dungeonSeed = std::abs(static_cast<int32_t>(state));
+		for (unsigned i = 0; i < seedTable.size(); ++i) {
+			state = advanceRng(state);
+			setLevelState(i, state);
 		}
 	}
 
-	GameState(uint8_t level, uint32_t seed, bool useNegatedState = false)
+	GameState(uint8_t level, uint32_t seed)
 	{
-		seedTable[level] = seed;
+		setLevelState(level, seed);
 
-		uint32_t state = useNegatedState ? -static_cast<int32_t>(seed) : seed;
+		uint32_t state = seed;
 		for (int i = level - 1; i >= 0; --i) {
-			state        = backtrackRng(state);
-			seedTable[i] = std::abs(static_cast<int32_t>(state));
+			state = backtrackRng(state);
+			setLevelState(i, state);
 		}
 		startingSeed = backtrackRng(state);
 
-		state = useNegatedState ? -static_cast<int32_t>(seed) : seed;
+		state = seed;
 		for (int i = level + 1; i < seedTable.size(); ++i) {
-			state        = advanceRng(state);
-			seedTable[i] = std::abs(static_cast<int32_t>(state));
+			state = advanceRng(state);
+			setLevelState(i, state);
 		}
 	}
+
+	void setLevelState(size_t level, uint32_t state)
+	{
+		stateTable[level] = static_cast<int32_t>(state);
+		seedTable[level]  = std::abs(stateTable[level]);
+	}
 };
 
 static void renderSeedTable(const GameState &state)
 {
 	time_t startingTime = std::chrono::system_clock::to_time_t(std::chrono::time_point<std::chrono::system_clock>(std::chrono::seconds(state.startingSeed)));
-	std::cout << "Game seed: " << state.startingSeed << " (" << std::put_time(std::gmtime(&startingTime), "%Y-%m-%d %H:%M:%S");
-	if (315532800 <= state.startingSeed && state.startingSeed <= 2177452799U) {
-		std::cout << " can be reached by setting the system time)\n";
+	std::cout << "Game seed: " << state.startingSeed << " (" << std::put_time(std::gmtime(&startingTime), "%Y-%m-%d %H:%M:%S") << " ";
+	if (state.startingSeed == -1) {
+		std::cout << "is the default game seed before 1970/after 2038";
```

Comment:
I was tempted to point this out to you on chat, but thought you might pick up on it without :)